src/main/java/chess/domain/chessboard/ChessBoard.java

```diff
@@ -1,50 +1,113 @@
+import chess.domain.position.File;
 import chess.domain.position.Position;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 public class ChessBoard {
 
     private final Map<Position, Piece> chessBoard;
 
-    private ChessBoard(Map<Position, Piece> chessBoard) {
+    public ChessBoard(Map<Position, Piece> chessBoard) {
         this.chessBoard = chessBoard;
     }
 
-    public static ChessBoard initializeChessBoard() {
-        return new ChessBoard(ChessBoardGenerator.initializeBoard());
-    }
+    public GameStatus move(Position source, Position target, GameStatus gameStatus) {
+        checkEmpty(source);
 
-    public void move(Position source, Position target) {
-        Piece piece = chessBoard.get(source);
-        Piece targetPiece = chessBoard.get(target);
+        Piece piece = getPiece(source);
+        piece.checkValidMove(gameStatus);
 
-        checkTargetIsTeam(piece, targetPiece);
-        piece.findRoute(source, target, targetPiece)
+        checkTargetIsTeam(piece, target);
+        piece.findRoute(source, target, isEmpty(target))
                 .forEach(this::checkObstacle);
 
-        replacePieceToTarget(source, target, piece);
+        return replacePieceToTarget(source, target, gameStatus);
+    }
+
+    private void checkEmpty(Position position) {
+        if (isEmpty(position)) {
+            throw new IllegalArgumentException("해당 공간에는 기물이 존재하지 않습니다.");
+        }
     }
 
     private void checkObstacle(Position position) {
-        if (!chessBoard.get(position)
-                .isEmpty()) {
-            throw new IllegalArgumentException("이동할 수 없습니다.");
+        if (!isEmpty(position)) {
+            throw new IllegalArgumentException("방해물이 있어 이동할 수 없습니다.");
         }
     }
 
-    private void checkTargetIsTeam(Piece source, Piece target) {
-        if(source.isTeam(target)) {
-            throw new IllegalStateException("같은 팀이 있는 곳으로는 이동할 수 없습니다.");
+    private void checkTargetIsTeam(Piece source, Position targetPosition) {
+        if (!isEmpty(targetPosition) && source.isTeam(getPiece(targetPosition))) {
+            throw new IllegalArgumentException("같은 팀이 있는 곳으로는 이동할 수 없습니다.");
         }
     }
 
-    private void replacePieceToTarget(Position source, Position target, Piece piece) {
-        chessBoard.put(source, new Empty());
+    private GameStatus replacePieceToTarget(Position source, Position target, GameStatus turn) {
+        if (!isEmpty(target) && getPiece(target).isKing()) {
+            return GAME_OVER;
+        }
+        Piece piece = getPiece(source);
+        chessBoard.remove(source);
         chessBoard.put(target, piece);
+
+        return turn.changeTurn();
+    }
+
+    public Map<Team, Score> calculateTotalScore() {
+        Map<Team, Score> score = new HashMap<>();
+        score.put(WHITE, calculateTeamScore(WHITE));
+        score.put(BLACK, calculateTeamScore(BLACK));
+
+        return score;
```

Comment:
```suggestion
        return Map.of(
                WHITE, calculateTeamScore(WHITE),
                BLACK, calculateTeamScore(BLACK)
        );
```
가 좀더 불변성을 유지하는 안정적인 코드일 것 같아요.